operator-common/src/main/java/io/strimzi/operator/common/model/PemTrustSet.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the set of certificates to be trusted by a TLS client or server
+ */
+public class PemTrustSet {
+    private static final String CERT_SUFFIX = ".crt";
+    private final Map<String, byte[]> trustedCertificateMap;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * Constructs the PemTrustSet. This also validates the provided Secret contains valid certificates.
+     * @param secret Kubernetes Secret containing the trusted certificates
+     */
+    public PemTrustSet(Secret secret) {
+        if (secret == null || secret.getData() == null) {
+            throw new RuntimeException("Cannot extract trust set from null secret");
+        }
+        Optional.of(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
```

Comment:
Kube resources should always have their metadata. So I think we should not do this just if present if we really need this.

Reply:
I've removed this